google/cloud/bigtable/tests/admin_integration_test.cc

```diff
@@ -231,6 +233,71 @@ TEST_F(AdminIntegrationTest, CreateListGetDeleteTableTest) {
   EXPECT_EQ(0, table_count);
 }
 
+/// @test Verify that `bigtable::TableAdmin` CheckConsistency works as expected.
+TEST_F(AdminIntegrationTest, CheckConsistencyIntegrationTest) {
+  using GC = bigtable::GcRule;
+  using namespace google::cloud::testing_util::chrono_literals;
+
+  std::string id =
+      "it-" + google::cloud::internal::Sample(
+                  generator_, 8, "abcdefghijklmnopqrstuvwxyz0123456789");
+  auto project_id = bigtable::testing::TableTestEnvironment::project_id();
+
+  auto instance_admin_client = bigtable::CreateDefaultInstanceAdminClient(
+      project_id, bigtable::ClientOptions());
+  bigtable::InstanceAdmin instance_admin(instance_admin_client);
+
+  bigtable::InstanceId instance_id(id);
+  bigtable::DisplayName display_name("Integration Tests " + id);
+
+  // Replication needs at least two clusters
+  auto cluster_config_1 =
+      bigtable::ClusterConfig("us-central1-b", 3, bigtable::ClusterConfig::HDD);
+  auto cluster_config_2 =
+      bigtable::ClusterConfig("us-central1-a", 3, bigtable::ClusterConfig::HDD);
+  bigtable::InstanceConfig config(
+      instance_id, display_name,
+      {{id + "-c1", cluster_config_1}, {id + "-c2", cluster_config_2}});
+
+  auto instance = instance_admin.CreateInstance(config).get();
+
+  // need to create table_admin with longer polling as ConsistencyCheck takes a
+  // little long
+  std::shared_ptr<bigtable::AdminClient> admin_client =
+      bigtable::CreateDefaultAdminClient(
+          bigtable::testing::TableTestEnvironment::project_id(),
+          bigtable::ClientOptions());
+  bigtable::TableAdmin table_admin(
+      admin_client, id,
+      bigtable::GenericPollingPolicy<bigtable::LimitedTimeRetryPolicy,
```

Comment:
I think the defaults are more generous than the values you provide, what exactly breaks when you do not set the policies?

Reply:
I think it is due to a bug in the code, unfortunately introduced by me...
`RPCPolicyParameters` is define as follows ([rpc_policy_parameters.h](https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/b1264a464b0dd36a0f103de68b1bbdc54bb6f7b9/google/cloud/bigtable/internal/rpc_policy_parameters.h#L28))
```
struct RPCPolicyParameters {
  std::chrono::milliseconds initial_delay;
  std::chrono::milliseconds maximum_delay;
  std::chrono::milliseconds maximum_retry_period;
};
```
but auto generated constants jumbles up value ([rpc_policy_parameters.inc](https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/b1264a464b0dd36a0f103de68b1bbdc54bb6f7b9/google/cloud/bigtable/internal/rpc_policy_parameters.inc#L64))
```
RPCPolicyParameters const kBigtableTableAdminLimits = {
    GOOGLE_CLOUD_CPP_BIGTABLETABLEADMIN_DEFAULT_INITIAL_DELAY,
    GOOGLE_CLOUD_CPP_BIGTABLETABLEADMIN_DEFAULT_MAXIMUM_RETRY_PERIOD,
    GOOGLE_CLOUD_CPP_BIGTABLETABLEADMIN_DEFAULT_MAXIMUM_DELAY,
};
```
Due to this, maximum delay becomes maximum retry and test case fails after a minute. 
I will change the code and test without changing polling policy.

Reply:
Sounds like we should fix this first and then come back to this PR, right?